### PR TITLE
Task 3 - Serverless API

### PR DESCRIPTION
**Pull request by abachyla**
1. [Task](https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/3_serverless_api/task.md)
2. Links
getProductsList API:  https://myln05b7e4.execute-api.eu-west-1.amazonaws.com/products
getProductsById API:  https://myln05b7e4.execute-api.eu-west-1.amazonaws.com/products/{productId}
FE Cloudfront: https://d14axewerpas2.cloudfront.net
[OpenAPI doc](https://app.swaggerhub.com/apis/EllenDanilowa/ProductServiceAPI/1)
3. FE PR: https://github.com/EllenDanilowa/shop-angular-cloudfront/pull/3
4. Done 10.07.2022 / Deadline 17.07.2022
5. Score 11/11

- [x] **1** - Product-service serverless config contains configuration for 2 lambda functions, API is working, but YAML configuration is correct
- [x] **2** - The getProductsList OR getProductsById lambda function returns a correct response (POINT1)
- [x] **3** - The getProductsById AND getProductsList lambda functions return a correct response code (POINT2)
- [x] **4** - Your own Frontend application is integrated with product service (`/products` API) and products from product-service are represented on Frontend. AND POINT1 and POINT2 are done.

Additional:
- [x] **+1** - Async/await is used in lambda functions
- [x] **+1** - ES6 modules are used for product-service implementation
- [x] **+1** - Webpack is configured for product-service
- [x] **+1** **(All languages)** - SWAGGER documentation is created for product-service
- [x] **+1** **(All languages)** - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)
- [x] **+1** **(All languages)** - Lambda handlers (`getProductsList`, `getProductsById`) code is written not in 1 single module (file) and separated in codebase.
- [x] **+1** **(All languages)** - Main error scenarious are handled by API ("Product not found" error).

### CODE REVIEW

**Comment by caglagezgen** [comment]
Good Job! Each task is as expected.

**Comment by VladislavLobakh** [comment]
Pay attention https://d14axewerpas2.cloudfront.net/admin/products/7567ec4b-b10c-48c5-9345-fc73c48a80aa is not filled with data

File: product-service/handlers/getProductsById.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const product = products.find((item) => item.id === id);
+
+  if (product) {
+    resolve(product);
+  } else {
+    reject(ERROR_TYPES.NOT_FOUND);
+  }
+});
+
+export const getProductsById = async (event = {}) => {
```
Comment (VladislavLobakh) [inline product-service/handlers/getProductsById.js:6]: getProductById

File: product-service/handlers/getProductsById.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  let response;
+
+  try {
+    const id = event.pathParameters && event.pathParameters.productId;
+    const product = await getProduct(id);
```
Comment (VladislavLobakh) [inline product-service/handlers/getProductsById.js]: you should have logic to receive products in a single place (service/controller)

**Review by VladislavLobakh — APPROVED** [review]
some minor marks (11/11)

**Comment by abachyla** [comment]
> Pay attention https://d14axewerpas2.cloudfront.net/admin/products/7567ec4b-b10c-48c5-9345-fc73c48a80aa is not filled with data

Thank you! According to the task requirements, it shouldn't be connected for now.

Task 3.2 4.This endpoint is not needed to be integrated with Frontend right now.

File: product-service/handlers/getProductsById.js (hunk 1 above)
Comment (abachyla) [reply product-service/handlers/getProductsById.js:6]: Totally agree, it makes more sense, but according to the task requirements it should be called getProductsById

Comment (VladislavLobakh) [reply product-service/handlers/getProductsById.js:6]: have a look nice example here:
https://github.com/VarvaraZadnepriak/product-service-updated-demo

**Pull request merged**